### PR TITLE
[Nano] Add how-to guide for TensorFlow Keras inference using bf16 mixed precision

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Add how-to guide for TensorFlow Keras inference using bf16 mixed precision

### How to test?
- [x] Document test: https://yuwentestdocs.readthedocs.io/en/nano-howto-tf-bf16-infer/doc/Nano/Howto/Inference/TensorFlow/tensorflow_inference_bf16.html
- [x] Github preview: https://github.com/Oscilloscope98/BigDL/blob/nano-howto-tf-bf16-infer/python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb
- [x] Local test on CPX-3, and hardware requirements verification on ICX-5, and platform without AVX512

### CODE REVIEW

File: python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+       "# Use BFloat16 Mixed Precision for TensorFlow Keras Inference"
+      ]
+     },
+     {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+       "Brain Floating Point Format (BFloat16) is a custom 16-bit floating point format designed for machine learning. BFloat16 is comprised of 1 sign bit, 8 exponent bits, and 7 mantissa bits. With the same number of exponent bits, BFloat16 has the same dynamic range as FP32, but requires only half the memory usage.\n",
+       "\n",
+       "BFloat16 Mixed Precison combines BFloat16 and FP32 during training, which could lead to increased performance and reduced memory usage. Compared to FP16 mixed precision, BFloat16 mixed precision has better numerical stability.\n",
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb:23]: may change `during training` to `during training and inference`

File: python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+       "> We recommend to run the commands above, especially `source bigdl-nano-init` before jupyter kernel is started, or some of the optimizations may not take effect."
+      ]
+     },
+     {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+       "> ⚠️ Warning\n",
+       ">\n",
+       "> BigDL-Nano will enable intel’s oneDNN optimizations by default. ONEDNN BFloat16 are only supported on platforms with AVX512.\n",
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb:66]: may unify `oneDNN` and `ONEDNN` and change `AVX512` to `AVX512 instruction set`

File: python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+       "import numpy as np\n",
+       "import time\n",
+       "\n",
+       "test_data = np.random.rand(32, 224, 224, 3)\n",
+       "\n",
+       "# FP32 inference\n",
+       "st1 = time.time()\n",
+       "for _ in range(100):\n",
+       "    fp32_model(test_data)\n",
+       "print(f'The time for 100 iterations of FP32 inference is: {time.time() - st1} ms')\n",
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb:216]: The unit of `time.time()` is second(s), not ms

File: python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+       "> Please also note that, when you have a custom model to quantize (e.g. inherated from `tf.keras.Model`), you need to specify the `input_spec` parameter to let OpenVINO accelerator know the shape of the model input.\n",
+       ">\n",
+       "> Please refer to [API documentation](https://bigdl.readthedocs.io/en/latest/doc/PythonAPI/Nano/tensorflow.html#bigdl.nano.tf.keras.InferenceOptimizer.quantize) for more information on `InferenceOptimizer.quantize`."
+      ]
+     },
+     {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+       "After quantizing your model with or without extra accelerator, you could then conduct BF16 mixed precision inferece as normal:"
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb:188]: A typo `inferece`

**Review by MeouSker77 — APPROVED** [review]
others LGTM

File: python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb (hunk 4 above)
Comment (Oscilloscope98) [reply python/nano/tutorial/notebook/inference/tensorflow/tensorflow_inference_bf16.ipynb]: Thank you for the comments :) I have fixed all of them.

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Pull request merged**